### ISSUE TITLE
lint unused (clippy) allow attributes

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3122]
````rust
fn main() {
    #[cfg_attr(feature = "cargo-clippy", allow(useless_format))]
    let _ = String::from("hello");
}
````

It would be nice to have a message here that ````allow(uselss_format)```` is itself not effective and can be removed.
Basically I'd like to have unused clippy warning suppression linted. :)

**Comment by EdJoPaTo** [comment]
I am used to having this on JavaScript / TypeScript with [no-unused-disable](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-unused-disable.md)

unused is probably not a well chosen name for Rust / Clippy as it already has a different meaning. Still I think it could be inspiring.

**Comment by xFrednet** [comment]
Implementing a lint for this is sadly rather difficult. [RFC 2383](https://rust-lang.github.io/rfcs/2383-lint-reasons.html) has proposed adding a `#[expect]` attribute that allows lints until they no longer are emitted. The implementation is sadly not trivial and held up on some implementation details. See rust-lang/rust#54503 and my implementation draft in rust-lang/rust#87835 if you want to know the current state of the implementation. Having a feature like this would definitely help a lot!

**Comment by y21** [comment]
Now that `#[expect]` has been stabilized I think we can close this :)
